controllers/loginRegisterController.js

```diff
@@ -1,25 +1,66 @@
 const express = require('express')
 const router = express.Router()
-const userService = require('../services/userService')
 const userDao = require('../dao/userDao')
 const jwtService = require('../services/jwtService')
 require('dotenv').config({ path: 'config/.env'})
 
+const{onClientError} = require("./errorHandler");
+
+
 router.use(express.json())
 
-router.post('/register', userService.createUser, (req, res) => {
-    res.sendStatus(200)
-})
-
-router.post('/login', async (req, res) => {
-    const {username, password} = req.body
-    if(await userService.authenticate(username, password)) {
-        const user = await userDao.getUserByUsername(username)
-        const token = jwtService.generateToken(user)
-        res.json({token})
-    } else {
-        res.sendStatus(403)
-    }
-})
-
-module.exports = router
\ No newline at end of file
+router.post('/register', (req, res) => {
+
+    userDao.createUser(req.body, false).then(user_r => {
+        if(user_r.success){
+            let created_user = user_r.user;
+            res.status(201).json({
+                "detail": "Created new user",
+                "user_id": created_user.id,
+                "username": created_user.username,
+            });
+        }
+        else{
+            onClientError(res, user_r.status_code, user_r.message);
+        }
+    }).catch(err => onServerError(res, err));
+});
+
+
+
+router.post("/login", (req, res) => {
+
+    jwtService.login(req.body).then(token_r=>{
+        if(token_r.success){
+            res.status(200).json({
+                "token": token_r.token,
+                "admin": token_r.admin
+            });
+        }
+        else{
+            onClientError(res, token_r.status_code, token_r.message);
+        }
+    }).catch(err => onServerError(res, err));
+});
+
+router.get("/my_account", jwtService.verifyToken, (req, res)=>{
```

Comment:
Can we maybe return user data in `router.post("/login")` after status code 200? and not using additional endpoint '/my_account' WDYT? Is there any condition that we need to make another request to get data? 🤔